examples/cpp/dds/DiscoveryServerExample/DiscoveryServer_main.cpp

```diff
@@ -186,7 +186,7 @@ int main(
                 }
 
                 case optionIndex::CONNECTION_PORT:
-                    connection_port = strtol(opt.arg, nullptr, 10);
+                    connection_port = static_cast<uint16_t>strtol(opt.arg, nullptr, 10);
```

Comment:
```
connection_port = static_cast<uint16_t>(strtol(opt.arg, nullptr, 10));
```
Please, compile in local before upload a commit

Same error is repeated in other casts